test-case/check-alsabat.sh

```diff
@@ -64,11 +73,84 @@ function __upload_wav_file
     done
 }
 
+function set_pga_to_unity_gain ()
+{
+    tmp=$(amixer controls | grep "$pga" | grep Volume)
+    search="name="
+    cname=${tmp#*$search}
+
+    # Get volume min and step to compute value for cset
+    # for 0 dB gain. The amixer line looks like
+    # "| dBscale-min=-50.00dB,step=1.00dB,mute=1"
+    scale=$(amixer cget name="$cname" | grep "dBscale" || true)
+    search="dBscale-min="
+    tmp=${scale#*$search}
+    min_db="${tmp%%dB*}"
+    search="step="
+    tmp=${scale#*$search}
+    step_db="${tmp%%dB*}"
+
+    # Get multiplied by 100 values by removing decimal dot
+    min_x100="${min_db//.}"
+    step_x100="${step_db//.}"
+    val=$(printf %d "$(((-min_x100) / step_x100))")
+
+    # Apply the computed value for requested gain
+    amixer cset name="$cname" "$val"
+}
+
+function set_pgas_list_to_unity_gain ()
+{
+    if [[ -z "$1" ]]; then
+	return
+    fi
+
+    for pga in $1; do
+	dlogi "Set $pga"
+	set_pga_to_unity_gain "$pga"
+    done
+}
+
+function get_snd_base ()
+{
+    tmp=${1#*"hw:"}
+    ncard="${tmp%%,*}"
+    ndevice="${tmp#*,}"
```

Comment:
@marc-hb Sure!

@aiChaoSONG This e.g. hw:1,0 conversion to /dev/snd/PCMC0D0p is because I'm not able to filter with sof-tplgtool.py with search -f "dev:hw:1,0". Maybe it's a quoting issue by me? These functions get simpler if such search can be done.